### PR TITLE
peering: intentions list test

### PR DESCRIPTION
**Pull request by acpana**
### Description
This diff adds a test to prove that the http api can list intentions for peered services. 

#### links

* [asana](https://app.asana.com/0/1201631904755166/1202439043013562/f)

### CODE REVIEW

File: agent/intentions_endpoint_test.go [hunk 1]
```diff
@@ -62,6 +62,35 @@ func TestIntentionList(t *testing.T) {
 			ids = append(ids, reply)
 		}
 
+		// TODO(peering): when we handle Upserts, we can use the for loop above. But it may be that we
+		// rip out legacy intentions before supporting that use case so run a config entry request instead here.
+		testutil.RunStep(t, "set up peered intention", func(t *testing.T) {
```
Comment (kisunji) [inline agent/intentions_endpoint_test.go:67]: this shouldn't be in a subtest; it's setup code that belongs in the same level as the writes above. You could alternatively wrap it in braces to distinguish the setup from the actual assertion tests
```
// SETUP
{
...
}
```

File: agent/intentions_endpoint_test.go [hunk 2]
```diff
@@ -71,21 +100,23 @@ func TestIntentionList(t *testing.T) {
 		require.NoError(t, err)
 
 		value := obj.(structs.Intentions)
-		require.Len(t, value, 4)
+		require.Len(t, value, 5)
 
-		require.Equal(t, []string{"bar->db", "foo->db", "zim->gir", "*->db"},
+		require.Equal(t, []string{"bar->db", "foo->db", "zim->gir", "peered->bar", "*->db"},
```
Comment (kisunji) [inline agent/intentions_endpoint_test.go:105]: We probably want an assertion for the `Peer` field as well

**Review by kisunji — COMMENTED** [review]
Some minor comments but looks good!

File: agent/intentions_endpoint_test.go (hunk 1 above)
Comment (acpana) [reply agent/intentions_endpoint_test.go]: ✅

**Review by kisunji — APPROVED** [review]

**Pull request merged**